src/Compilers/VisualBasic/Portable/Parser/ParseConditional.vb

```diff
@@ -377,12 +380,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
         Private Function ParseWarningDirective(hashToken As PunctuationSyntax) As DirectiveTriviaSyntax
             Debug.Assert(CurrentToken.Kind = SyntaxKind.IdentifierToken,
-                         "ParseWarningDirective called with token that is not an IdentifierToken")
+                         NameOf(ParseWarningDirective) & " called with token that is not an {NameOf(SyntaxKind.IdentifierToken)}")
```

Comment:
It looks like ```{NameOf(SyntaxKind.IdentifierToken)}``` is going to be in the resulting string, verbatim. Another concatenation should be used instead.